compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -575,7 +576,9 @@ public void visit(BLangTransaction transactionNode) {
         int previousCommitCount = this.commitCount;
         int previousRollbackCount = this.rollbackCount;
         boolean prevCommitRollbackAllowed = this.commitRollbackAllowed;
+        boolean prevWithinTransactionBlock = this.withinTransactionBlock;
```

Comment:
We already have a flag, called withinTransactionScope, I think we can use it

Reply:
Fixed